OTAnalytics/plugin_prototypes/eventlist_exporter/eventlist_exporter.py

```diff
@@ -0,0 +1,168 @@
+    ) -> None:
+        df_events = EventListDataFrameBuilder(events=events, sections=sections).build()
+        df_sections = SectionsDataFrameBuilder(sections=sections).build()
+        self._write_to_excel(file, df_events, df_sections)
+
+    def _write_to_excel(
+        self, file: Path, df_events: pd.DataFrame, df_sections: pd.DataFrame
+    ) -> None:
+        writer = pd.ExcelWriter(file, engine="openpyxl")
+        df_events.to_excel(writer, index=False, sheet_name="Events")
+        df_sections.to_excel(writer, index=False, sheet_name="Sections")
+        writer.close()
+
+    def get_extension(self) -> str:
+        return "xlsx"
+
+    def get_name(self) -> str:
+        return OTC_EXCEL_FORMAT_NAME
+
+
+class EventListCSVExporter(EventListExporter):
+    def export(
+        self, events: Iterable[Event], sections: Iterable[Section], file: Path
+    ) -> None:
+        df_events = EventListDataFrameBuilder(events=events, sections=sections).build()
+        self._write_to_excel(file, df_events)
+
+    def _write_to_excel(self, file: Path, df_events: pd.DataFrame) -> None:
+        df_events.to_csv(file, index=False)
+
+    def get_extension(self) -> str:
+        return "csv"
+
+    def get_name(self) -> str:
+        return OTC_CSV_FORMAT_NAME
+
+
+class EventListDictPrinter(EventListExporter):
+    def export(
+        self, events: Iterable[Event], sections: Iterable[Section], file: Path
+    ) -> None:
+        print("Events:")
+        print([event.to_dict() for event in events])
+        print("__________________________________________________________________")
+        print("Sections:")
+        print([section.to_dict() for section in sections])
+
+    def get_extension(self) -> str:
+        return ""
+
+    def get_name(self) -> str:
+        return "print dicts to console"
+
+
+class EventListDataFramePrinter(EventListExporter):
+    def export(
+        self, events: Iterable[Event], sections: Iterable[Section], file: Path
+    ) -> None:
+        df_events = EventListDataFrameBuilder(events=events, sections=sections).build()
+        df_sections = SectionsDataFrameBuilder(sections=sections).build()
+        print("Events:")
+        print(df_events)
+        print("__________________________________________________________________")
+        print("Sections:")
+        print(df_sections)
+
+    def get_extension(self) -> str:
+        return ""
+
+    def get_name(self) -> str:
+        return "print Dataframe to console"
+
+
+class ExporterNotFoundError(Exception):
+    pass
+
+
+@dataclass
+class EventListExporterStrategies:
+    AVAILABLE_EXPORTERS: list[EventListExporter] = field(
+        default_factory=lambda: [
+            EventListExcelExporter(),
+            EventListCSVExporter(),
+            # EventListPrinter(),
+            # EventListDataFramePrinter(),
+        ]
+    )
+
+    def get_exporter_by_name(self, name: str) -> EventListExporter:
+        for exporter in self.AVAILABLE_EXPORTERS:
```

Comment:
Store the available exporters in a dictionary and use `...get(name)` to get one.